src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -4,19 +4,28 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
 using System.Formats.Cbor;
 
 namespace System.Security.Cryptography.Cose
 {
+    /// <summary>
+    /// Represents a bucket of header parameters of a COSE message.
+    /// </summary>
     public sealed class CoseHeaderMap : IDictionary<CoseHeaderLabel, CoseHeaderValue>, IReadOnlyDictionary<CoseHeaderLabel, CoseHeaderValue>
     {
         private static readonly CoseHeaderMap s_emptyMap = new CoseHeaderMap(isReadOnly: true);
 
         private readonly Dictionary<CoseHeaderLabel, CoseHeaderValue> _headerParameters = new Dictionary<CoseHeaderLabel, CoseHeaderValue>();
 
+        /// <summary>
+        /// Gets a value that indicates whether the map cannot be modified.
+        /// </summary>
+        /// <remarks>A read-only map is a protected map of a decoded COSE message.</remarks>
```

Comment:
The basis and conclusion feel backwards here.  We might add other places where maps are read-only, but protected always will be.

Personally, I'd delete the remark, but it can also be

```suggestion
        /// <remarks>The "protected headers" collection in a COSE message is always read-only.</remarks>
```

Reply:
Since that's the only case where IsReadOnly == true, I thought it was worth pointing it out, otherwise users might be scratching their heads thinking when the thing is read-only.